packages/browser/__tests__/ClientAuthentication.spec.ts

```diff
@@ -318,4 +402,243 @@ describe("ClientAuthentication", () => {
+        sessionInfoManager: mockSessionInfoManager(mockedStorage),
+      });
+
+      await expect(clientAuthn.getCurrentIssuer()).resolves.toBeNull();
+    });
+
+    it("returns null if the issuer's JWKS isn't available", async () => {
+      const sessionId = "mySession";
+      // @ts-ignore
+      window.localStorage = new LocalStorageMock({
+        [KEY_CURRENT_SESSION]: sessionId,
+      });
+      const mockedIssuerConfig = mockIssuerConfigFetcher({
+        jwksUri: "https://some.issuer/jwks",
+      } as IIssuerConfig);
+      const mockedStorage = await mockSessionStorage(sessionId);
+      mockFetch(new Response("Not a valid JWKS"));
+
+      const clientAuthn = getClientAuthentication({
+        issuerConfigFetcher: mockedIssuerConfig,
+        sessionInfoManager: mockSessionInfoManager(mockedStorage),
+      });
+
+      await expect(clientAuthn.getCurrentIssuer()).resolves.toBeNull();
+    });
+
+    it("returns null if the current issuer doesn't match the ID token's", async () => {
+      const sessionId = "mySession";
+      // @ts-ignore
+      window.localStorage = new LocalStorageMock({
+        [KEY_CURRENT_SESSION]: sessionId,
+      });
+      const mockedIssuerConfig = mockIssuerConfigFetcher({
+        jwksUri: "https://some.issuer/jwks",
+      } as IIssuerConfig);
+      const mockedStorage = await mockSessionStorage(
+        sessionId,
+        mockIdTokenPayload(
+          "https://my.pod/profile#me",
+          "https://some-other.issuer",
+          "https://some.app/registration"
+        )
+      );
+      mockFetch(new Response(JSON.stringify({ keys: [mockJwk()] })));
+
+      const clientAuthn = getClientAuthentication({
+        issuerConfigFetcher: mockedIssuerConfig,
+        sessionInfoManager: mockSessionInfoManager(mockedStorage),
+      });
+
+      await expect(clientAuthn.getCurrentIssuer()).resolves.toBeNull();
+    });
+
+    it("returns null if the current client ID doesn't match the ID token audience", async () => {
+      const sessionId = "mySession";
+      // @ts-ignore
+      window.localStorage = new LocalStorageMock({
+        [KEY_CURRENT_SESSION]: sessionId,
+      });
+      const mockedIssuerConfig = mockIssuerConfigFetcher({
+        jwksUri: "https://some.issuer/jwks",
+      } as IIssuerConfig);
+      const mockedStorage = await mockSessionStorage(
+        sessionId,
+        mockIdTokenPayload(
+          "https://my.pod/profile#me",
+          "https://some.issuer",
+          "https://some-other.app/registration"
+        )
+      );
+      mockFetch(new Response(JSON.stringify({ keys: [mockJwk()] })));
+
+      const clientAuthn = getClientAuthentication({
+        issuerConfigFetcher: mockedIssuerConfig,
+        sessionInfoManager: mockSessionInfoManager(mockedStorage),
+      });
+
+      await expect(clientAuthn.getCurrentIssuer()).resolves.toBeNull();
+    });
+
+    it("returns null if the ID token signature cannot be verified", async () => {
```

Comment:
And here I'm having real trouble figuring out what the reason is that the ID token signature cannot be verified - though that might just be my ignorance of OIDC.

Reply:
I think the test description wasn't great either. 26a099b4adc9a94574af9c7c6ad816c29cbb0ff5 should make it better